Sources/NIO/EventLoopFuture.swift

```diff
@@ -603,6 +605,33 @@ extension EventLoopFuture {
         return next.futureResult
     }
 
+    /// When the current `EventLoopFuture<Value>` is fulfilled, run the provided callback, which
+    /// performs a synchronous computation and returns either a new value (of type `NewValue`) or
+    /// an error depending on the `Result` returned by the closure.
+    ///
+    /// Operations performed in `flatMapResult` should not block, or they will block the entire
+    /// event loop. `flatMapResult` is intended for use when you have a data-driven function that
+    /// performs a simple data transformation that can potentially error.
+    ///
+    ///
+    /// - parameters:
+    ///     - body: Function that will receive the value of this `EventLoopFuture` and return
+    ///         a new value or error lifted into a new `EventLoopFuture`.
+    /// - returns: A future that will receive the eventual value.
+    @inlinable
+    public func flatMapResult<NewValue, SomeError: Error>(file: StaticString = #file,
```

Comment:
this moves the flatmap from where all `reduce` were defined, to where all the `flatMap` are defined